### PR TITLE
Nodejs oic bindings

### PR DESCRIPTION
**Pull request by gabrielschulhof**
Also cleans up whitespace in Node.js bindings and adds invalidation of JS objects that wrap native pointers.

### CODE REVIEW

**Comment by gabrielschulhof** [comment]
I've modified the semaphore to build OIC, but I've kept the ```sed``` that removes DTLS from the configuration.

**Comment by barbieri** [comment]
OIC is under big changes to cope with different transports and some optimizations.

@ceolin, @glima and @otaviobp would you hint @gabrielschulhof on what may impact him?

**Comment by gabrielschulhof** [comment]
```
sol_oic_client_find_resource(client, NULL, "", resourceFound, NULL);
```
would be nice. Aside from that, the contents of ```sol_network_link_addr``` will probably fluctuate, and I'll have to keep up.

**Comment by gabrielschulhof** [comment]
If ```sol_oic_client_find_resource()``` accepted a ```NULL``` pointer for the ```struct sol_network_link_addr *``` parameter, I could treat ```struct sol_network_link_addr *``` pointers as opaque and simply wrap them in JS objects when I get them from soletta.

**Comment by gabrielschulhof** [comment]
That's how ```iotivity-node``` works.

**Comment by gabrielschulhof** [comment]
... in that the destination can be ```NULL```, I mean.

**Comment by gabrielschulhof** [comment]
Another idea: If ```struct sol_network_link_addr``` were a shallow structure, meaning it contained no pointers to other structures, I could also copy it by value and maintain the copies myself, like so:

```
struct sol_network_link_addr *destination = malloc(sizeof(
struct sol_network_link_addr));
*destination = *pointer_from_soletta;
```

Then, when the contents of the structure settled down, I could open it up and expose its properties to JS.

**Comment by ceolin** [comment]
@barbieri  @gabrielschulhof I'm changing **COAP** API separating the transport from the coap logic. It should not affect OIC API that much.

**Comment by gabrielschulhof** [comment]
This PR provides node.js bindings for ```sol_oic_client_*()``` and ```sol_oic_server*()``` as well as ```sol_network_addr_to_str()```.

File: bindings/nodejs/generate-main.awk [hunk 1]
```diff
@@ -0,0 +1,91 @@
+# This file is part of the Soletta Project
+#
+# Copyright (C) 2015 Intel Corporation. All rights reserved.
```
Comment (zqzhang) [inline bindings/nodejs/generate-main.awk]: `s/2015/2016/g` for all new added files

**Pull request closed without merging**